docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,203 @@
+---
+title: Keyboard handling
+sidebar_title: Keyboard handling
+description: A guide to handling common keyboard interactions.
+---
+
+Keyboard handling is crucial for creating an excellent user experience in your Expo app. This guide covers common keyboard interactions and how to manage them effectively.
+
+React Native provides several keyboard-related APIs, with [Keyboard](https://reactnative.dev/docs/keyboard) and [KeyboardAvoidingView](https://reactnative.dev/docs/keyboardavoidingview) being the most frequently used. For more complex or custom keyboard interactions, you might consider using [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller), a library that offers advanced keyboard handling capabilities.
+
+## The basics
+
+### Keyboard events
+
+The Keyboard module allows you to listen for native events, react to them, and make changes to the keyboard, such as dismissing it.
+
+To listen for keyboard events, use the `Keyboard.addListener` method. This method takes an event name and a callback function as arguments. The callback function will be called with the event data when the keyboard is shown or hidden.
+
+```js
```

Comment:
````suggestion
```tsx
````

For concistency, let's use `tsx` as our primary code language for all snippets in this guide. We've recently migrated a lot of code examples to use TypeScript since that's one of the common things user ask for.

Also, (to be redundant in the review), it'd be nice to add code file names for all code blocks in this guide. For reference on how, please see [this section](https://github.com/expo/expo/blob/main/docs/README.md#add-code-block) in Docs' readme.